### PR TITLE
Added table of contents to README.md

### PR DESCRIPTION
**Pull request by kunagpal**
* Corrected error in Newman library usage sample snippet
* Segregated sections using horizontal rules

> **TODO:**
- [x] Add custom template usage documentation
- [x] Added collection run reports for the [sample collection](https://github.com/postmanlabs/newman/blob/develop/examples/sample-collection.json)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,16 +10,46 @@ _Supercharge your API workflow<br/>Modern software is built on APIs. Postman hel
...
+
+5. [Using Newman with the Postman Cloud API](#using-newman-with-the-postman-cloud-api)
+
+6. [Communinty Support](#community-support)
+
+7. [License](#license)
+
+---
+
 ## IMPORTANT NOTICE
```
Comment (shamasis) [inline README.md]: this stays up

File: README.md [hunk 2]
```diff
@@ -10,16 +10,46 @@ _Supercharge your API workflow<br/>Modern software is built on APIs. Postman hel
...
-> To view documentation of current stable 2.x release of Newman, refer the
-> [Newman v2.x README](https://github.com/postmanlabs/newman/blob/release/2.x/README.md) of 2.x release.
+> For Newman v2.x release documentation, see the [Newman v2.x README](https://github.com/postmanlabs/newman/blob/release/2.x/README.md).
+
+---
 
 ## Getting started
 
-To run Newman, ensure that you have NodeJS >= v4. A copy of the NodeJS installable can be downloaded from [https://nodejs.org/en/download/package-manager/](https://nodejs.org/en/download/package-manager).
+To run Newman, ensure that you have NodeJS >= v4. A copy of the NodeJS installable can be downloaded from [here](https://nodejs.org/en/download/package-manager).
```
Comment (shamasis) [inline README.md]: revert to URL.

File: README.md [hunk 3]
```diff
@@ -164,15 +196,140 @@ To enable JSON reporter, provide `--reporters json` as a CLI option.
...
+HTML reporter, provide `--reporters html` as a CLI option.
 
 | CLI Option  | Description       |
 |-------------|-------------------|
 | `--reporter-html-export <path>` | Specify a path where the output HTML file will be written to disk. If not specified, the file will be written to `newman/` in the current working directory. |
+| `--reporter-html-template <path>` | Specify a path to the custom template which will be used to render the HTML report. This option depends on `--reporter html` and `--reporter-html-export` being present in the run command. If this option is not specified, the [default template](https://github.com/postmanlabs/newman/blob/develop/lib/reporters/html/template-default.hbs) is used |
+
+###### Custom HTML reporter templates
+
+Custom templates (currently handlebars only) can be passed to the HTML reporter with `--reporter-html-template <path>`, in conjunction with `--reporters html` and `--reporter-html-export` flags.
```
Comment (shamasis) [inline README.md]: shorten it. make it not have a heading.

File: README.md [hunk 4]
```diff
@@ -164,15 +196,140 @@ To enable JSON reporter, provide `--reporters json` as a CLI option.
...
+The [default template](https://github.com/postmanlabs/newman/blob/develop/lib/reporters/html/template-default.hbs) is used when the custom template option has been left out. The template receives
+a data object of the following format:
+
+```javascript
```
Comment (shamasis) [inline README.md]: move it to examples/reporters + add 2 more

File: README.md [hunk 5]
```diff
@@ -164,24 +197,27 @@ To enable JSON reporter, provide `--reporters json` as a CLI option.
...
+| `--reporter-html-template <path>` | Specify a path to the custom template which will be used to render the HTML report. This option depends on `--reporter html` and `--reporter-html-export` being present in the run command. If this option is not specified, the [default template](https://github.com/postmanlabs/newman/blob/develop/lib/reporters/html/template-default.hbs) is used |
+
+In addition, custom templates (currently handlebars only) can be passed to the HTML reporter with `--reporter-html-template <path>`, in conjunction with `--reporters html` and `--reporter-html-export` flags.
+The [default template](https://github.com/postmanlabs/newman/blob/develop/lib/reporters/html/template-default.hbs) is used when the custom template option has been left out.
 
 ##### JUNIT/XML reporter options
-Newman can output a summary of the collection run to a JUnit compatible XML file. To enable JSON reporter, provide
+Newman can output a summary of the collection run to a JUnit compatible XML file. To enable the JUNIT reporter, provide
 `--reporters html` as a CLI option.
```
Comment (czardoz) [inline README.md]: `--reporters junit`

File: README.md [hunk 6]
```diff
@@ -164,24 +197,27 @@ To enable JSON reporter, provide `--reporters json` as a CLI option.
...
-JSON reporter, provide `--reporters html` as a CLI option.
+The built-in HTML reporter produces and HTML output file outlining the summary and report of the Newman run. To enable the
+HTML reporter, provide `--reporters html` as a CLI option.
 
 | CLI Option  | Description       |
 |-------------|-------------------|
 | `--reporter-html-export <path>` | Specify a path where the output HTML file will be written to disk. If not specified, the file will be written to `newman/` in the current working directory. |
+| `--reporter-html-template <path>` | Specify a path to the custom template which will be used to render the HTML report. This option depends on `--reporter html` and `--reporter-html-export` being present in the run command. If this option is not specified, the [default template](https://github.com/postmanlabs/newman/blob/develop/lib/reporters/html/template-default.hbs) is used |
+
+In addition, custom templates (currently handlebars only) can be passed to the HTML reporter with `--reporter-html-template <path>`, in conjunction with `--reporters html` and `--reporter-html-export` flags.
```
Comment (czardoz) [inline README.md]: simplify

**Pull request merged**